test/parallel/test-assert.js

```diff
@@ -452,6 +452,24 @@ assert.throws(makeBlock(thrower, TypeError));
                      'a.doesNotThrow is not catching type matching errors');
 }
 
+// stack shouldn't start with assert's innerFail, but caller instead
+{
+  let threw = false;
+  try {
+    a.doesNotThrow(makeBlock(thrower, Error));
+  } catch (e) {
+    threw = true;
+    assert.ok(e instanceof a.AssertionError);
+    const lines = e.stack.split('\n')
+    assert.equal(lines.length, 11)
+    assert.ok(lines[0].includes('Got unwanted exception'))
+    assert.ok(lines[2].includes('Object.<anonymous>'))
+    assert.ok(lines[2].includes('test-assert.js'))
+  }
+  assert.strictEqual(true, threw,
+                     'a.doesNotThrow is not catching type matching errors');
+}
```

Comment:
Hm, this test is almost completely redundant to e.g. [1](https://github.com/nodejs/node/pull/18023/files#diff-81c8c31c5728ec9ed47700e55d9ceebfR492) and [2](https://github.com/nodejs/node/pull/18023/files#diff-81c8c31c5728ec9ed47700e55d9ceebfR446). The only untested thing here is that those tests do not make sure that the stack frames do not contain `at doesNotReject`. I would just add that after [line 449](https://github.com/nodejs/node/pull/18023/files#diff-81c8c31c5728ec9ed47700e55d9ceebfR449).